client/List.jsx

```diff
@@ -0,0 +1,68 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+class Item extends React.Component {
+    constructor(props){
+        super(props);
+        this.state = {isCompleted: false};
+        this.handleClick = this.handleClick.bind(this);
+    }
+    handleClick(){
+        this.setState({isCompleted: !this.state.isCompleted});
+    }
+    render(){
+        const {name} = this.props;
+        const itemWithStatus = this.state.isCompleted ? <strike>{name}</strike> : name; 
+        return (
+        <span onClick={this.handleClick}>
+            {itemWithStatus}
+        </span>
+        );
+    }
+};
+
+const List = (props) => {
```

Comment:
IMO you can completely remove this component.  I can see why you may want to use some encapsulation, but you don't need to go granular in this way.  A DynamicList component can handle the `<ul>`, mapping of list items, and state.  From `DynamicList` you can pass state handlers as props into the `<Item />` elements.

Reply:
make sense. Thanks